src/datasets/io/json.py

```diff
@@ -63,49 +64,84 @@ def __init__(
         dataset: Dataset,
         path_or_buf: Union[PathLike, BinaryIO],
         batch_size: Optional[int] = None,
+        num_proc: Optional[int] = None,
         **to_json_kwargs,
     ):
         self.dataset = dataset
```

Comment:
It would be nice to have an assertion that checks if `num_proc > 0`.
```suggestion
        assert num_proc is None or num_proc > 0, "num_proc must be an integer > 0."
        self.dataset = dataset
```